Vet clinic database: query and update animals table

ShoiraTa opened the pull request:
- Use database transactions.
- Modify and delete data in SQL.
- Prepare complex queries that answer analytical question

AshakaE commented on queries.sql:
```diff
@@ -5,4 +5,36 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pickachu';
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
```
- [x] Kindly verify that change was made before the `ROLLBACK` query

ShoiraTa replied:
Hello, could you please help me with telling how I can verify that?

AshakaE commented on queries.sql:
```diff
@@ -5,4 +5,36 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pickachu';
+UPDATE animals SET species = 'pokemon' WHERE species is NULL;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-1-1';
+SAVEPOINT delete_one;
+UPDATE animals SET weight_kg = -1 * weight_kg;
+ROLLBACK TO delete_one;
```
- [x] Kindly verify that changes were made before all `ROLLBACK` queries

AshakaE commented on queries.sql:
```diff
@@ -5,4 +5,36 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pickachu';
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species is NULL;
+COMMIT;
```
- [x] Kindly verify that change was made before and after the `COMMIT` query

AshakaE commented on queries.sql:
```diff
@@ -5,4 +5,36 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pickachu';
+-- 2
+SELECT AVG(weight_kg) FROM animals;
+-- 16.6488888888888889
+SELECT MAX(escape_attempts) FROM animals;
```
This query answers the question  `Who escapes the most, neutered or not neutered animals?` but it's not fully correct because it missing the last condition(neutered or not neutered)

- [x] Kindly update query

AshakaE commented on data.sql:
```diff
@@ -1,17 +1,40 @@
+
+INSERT INTO animals
+(name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES ('Angemon', DATE '12-06-2005', '1', 'true', '45');
```
- [x] Kindly use the correct value for weight_kg, I believe it's `-45.0`

AshakaE commented on data.sql:
```diff
@@ -1,17 +1,40 @@
\ No newline at end of file
+INSERT INTO animals
+(name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES ('Squirtle', DATE '02-04-1993', '3', 'false', '12.3');
```
- [x] Kindly use the correct value for weight_kg, I believe it's `-12.13`

AshakaE commented on data.sql:
```diff
@@ -1,17 +1,40 @@
+
+INSERT INTO animals
+(name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES ('Plantmon', DATE '11-15-2022', '2', 'true', '5.7');
```
- [x] Kindly use the correct value for weight_kg, I believe it's `-5.7`

AshakaE commented on data.sql:
```diff
@@ -1,17 +1,40 @@
+
+INSERT INTO animals
+(name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES ('Charmander', DATE '08-02-2020', '0', 'false', '11');
```
- [x] Kindly use the correct value for weight_kg, I believe it's `-11.0`

ShoiraTa commented:
- Added verifications that change was made before the ROLLBACK and commit queries
- Corrected negative values
- Updated query that answers the question `Who escapes the most, neutered or not neutered animals?`

SajjadAhmad14 reviewed (approved):
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Shoira, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Please leave any questions or comments in the PR if you have any doubts or unclarity!
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).
